pkg/identity/identitymanager/manager.go

```diff
@@ -143,6 +147,12 @@ func (idm *IdentityManager) GetIdentityModels() []*models.IdentityEndpoints {
 	return identities
 }
 
+func (idm *IdentityManager) subscribe(o Observer) {
+	idm.mutex.Lock()
+	defer idm.mutex.Unlock()
+	idm.observers = append(idm.observers, o)
```

Comment:
I'd prefer to use `map[Observer]struct{}` to ensure that we can't update a given observer multiple times, which could occur if we call `Subscribe` multiple times for the same observer. However, this ensures that an `Observer` is always a comparable type, which we can't guarantee at compile-time.

Reply:
This occurred to me, but with the minimal usage there is in this PR (exactly one static subscribe) I didn't see a strong motivation just yet. If you think we're likely to miss problems of this kind when extending this code in future then maybe I better make this change now.

Reply:
Addressed.